src/components/Gesture.svelte

```diff
@@ -40,6 +41,30 @@
 
   let isThisRecording = false;
 
+  let showCountdown = false;
+
+  const countdownInitialValue = 3;
+
+  let countdownValue = 3;
+
+  let countdownInterval: number = 500; // the countdown interval in milliseconds
+
+  // Countdown to collect data
+  async function countdownStart(): Promise<void> {
+    countdownValue = countdownInitialValue;
+    return new Promise<void>(resolve => {
+      const interval = setInterval(() => {
+        countdownValue --;
+          if (countdownValue === 0 && showCountdown === true) {
```

Comment:
This `showCountdown === true` check is not needed. We can assume if this function is being called, the countdown is being shown.
```suggestion
          if (countdownValue === 0) {
```